src/components/m-datepicker/js/datepicker.js

```diff
@@ -1,179 +1,81 @@
-      const year = this.datepickerBody.getAttribute('year');
-      const month = this.datepickerBody.getAttribute('month');
-      const dayAsValue = this.datepickerBody.getAttribute('value'); // new value - after click on a day
-      let dayAsDay = this.datepickerBody.getAttribute('day'); // prev value - prior click on a day
-      if (dayAsDay === 'false') {
-        dayAsDay = false;
-      }
-      const day = dayAsValue || dayAsDay;
-      const eventInit = { bubbles: true, cancelable: true, composed: true };
 
-      if (day) {
-        const choosenDate = new Date(year, month, day);
-
-        if (this.outputIso) {
-          fire(this.okButton, AXA_EVENTS.AXA_CLICK, {
-            value: choosenDate.getTime(),
-            button: OK,
-          }, eventInit);
-        } else {
-          fire(this.okButton, AXA_EVENTS.AXA_CLICK, {
-            value: choosenDate,
-            button: OK,
-          }, eventInit);
-        }
-      } else {
-        fire(this.okButton, AXA_EVENTS.AXA_CLICK, {
-          value: '',
-          button: OK,
-        }, eventInit);
+    this.onOkButtonListenerEnd = on(this.okButton, EVENTS.CLICK, () => {
+      let out = '';
+      const value = this.datepickerBody.getAttribute('date');
+      if (value) {
+        out = new Date(Date.parse(value));
+        fire(this.wcNode, 'date-changed', { value: out }, { bubbles: true, cancelable: true, composed: true });
       }
     });
   }
 
-  offListenToButtons() {
-    if (this.unCancelButtonListenerEnd) {
-      this.unCancelButtonListenerEnd();
-    }
-    if (this.unOkButtonListenerEnd) {
-      this.unOkButtonListenerEnd();
-    }
-  }
-
-  handleChangeDropdownMonth = (e) => {
+  handleChangeDropdownMonth(e) {
     e.preventDefault();
-    e.stopPropagation();
-    const { target } = e;
-    const month = target.getAttribute('value');
+    const month = e.detail;
+    if (month) {
+      const newDate = this.datepickerBody.getAttribute('date');
+      const parsedDate = new Date(Date.parse(newDate));
+      parsedDate.setMonth(month);
 
-    // or if a dropdown changes, update the datepicker body
-    if (month || month === 0) {
-      this.datepickerBody.setAttribute('month', month);
+      this.datepickerBody.setAttribute('date', parsedDate.toISOString());
     }
   }
 
-  handleChangeDropdownYear = (e) => {
+  handleChangeDropdownYear(e) {
     e.preventDefault();
-    e.stopPropagation();
-    const { target } = e;
-    const year = target.getAttribute('value');
-
-    if (year || year === 0) {
-      this.datepickerBody.setAttribute('year', year);
+    const year = e.detail;
+    if (year) {
+      const newDate = this.datepickerBody.getAttribute('date');
+      const parsedDate = new Date(Date.parse(newDate));
+      parsedDate.setFullYear(year);
+      this.datepickerBody.setAttribute('date', parsedDate.toISOString());
     }
   }
 
-  handleChangeDatepickerBody = (e) => {
```

Comment:
same here, please revert to class fields

Reply:
true. done.